### DISCUSSION TITLE
gcsfuse encrypt my files

### DISCUSSION THREAD

**Discussion by iulian-glavan** [GoogleCloudPlatform/gcsfuse#1724] [General]
Hi,
Do you know how to disable that `gcsfuse` to not encrypt my files anymore?
So i'm trying to mount the bucket into VM, i can see the structure of the bucket but it seems that content of the files are encrypted. 

And if use `gsutil` is not the case.
gcsfuse --implicit-dirs   -o rw -o allow_other -file-mode=775 -dir-mode=775 --only-dir folder bucket_dev ~/folder_DEV

And no error found in the debug/log.

**Comment by raj-prince** [comment]
GCSFuse reads the object with `content-encoding: gzip` in metadata as encrypted only.

Although if you want, there are two ways:
(a) You have to use older version of gcsfuse - before [this](https://github.com/GoogleCloudPlatform/gcsfuse/releases/tag/v1.1.0) - but this is not recommended.
(b) Download the object with `gsutil` or `gcloud` and upload them as un-compressed object. With this process, content encoding in metadata will change. After that gcsfuse will show all those objects as normal (uncompressed) object.

For more details see the `File transcoding` section of [this](https://cloud.google.com/storage/docs/gcs-fuse#expandable-1) doc.

**Comment by iulian-glavan** [comment]
yes (b) it seems to work, but for eg. if i want to push something into that bucket from github that it will encrypt my content again..
would be the problem that somehow github is encrypting my files when deploying new version of the scripts?

**Comment by raj-prince** [comment]
What utility are you using to upload the data to GCS bucket?
Are you using - https://github.com/google-github-actions/upload-cloud-storage?tab=readme-ov-file#inputs?

Anyway, to answer your question, if your content is pushed to bucket via github as encrypted then again you will get the encrypted data.

If you are using https://github.com/google-github-actions/upload-cloud-storage to upload the data to your bucket. Then you have to change `gzip: false` in your configuration to stop encrypting the data via github.

**Comment by iulian-glavan** [comment]
Hi,
You're right the files was encoding bcs of the gzip was true.
Was set to false and it's working :).
Thank you!
